### DISCUSSION TITLE
[CLI] css font-family utilities for ease of use and clarity

### DISCUSSION THREAD

**Discussion by NathanWalker** [NativeScript/rfcs#9] [2. Archives]
A sore spot for quite a long time is trying to get the font-family combination correct for iOS/Android.
For a custom font such as `barlow-bold.ttf` the css needs to be:
```
font-family: Barlow, barlow-bold;
```
Getting this correct is sometimes incredibly frustrating for custom fonts.

Proposal: Potentially add a convenient cli command which prints out exactly what css is needed for any given font:

```
$ ns font /path/to/barlow-bold.ttf

Here's the CSS to use:

font-family: Barlow, barlow-bold;

Also make sure you place the font inside the `fonts` folder at root of your app.
```

Benefit: Save ton of time debugging why custom fonts aren't working.

**Comment by NathanaelA** [comment]
I fixed the RFC, android does and iOS (I believe also) use case sensitive file systems.   So if the file name is actually all lowercase "barlow-bold.ttf", and the actual internal font name is "Barlow", so both have to be cased properly; the file linking part needs to be cased to lower case to match the file name so it can be loaded on case sensitive file systems.       This really proves why we need to have this feature; it is so easy for us to mess it up, I had to think it thru several times.  :grinning: 

---

Just as a forward direction; we can probably use code from:
https://www.npmjs.com/package/font-finder here to parse the OTF/TTF to get font name out of font.  (Specifically; https://github.com/princjef/font-finder/blob/master/src/parse.ts) as it parsed the TTF files to get the font into out we would need to parse.

---

As an change, maybe rather than a path to the font file; it is just "tns fonts" and it lists the css for all fonts already in the `fonts/` folder, so that way the developer knows up front that he forgot to copy the font to his app's font folder???

**Comment by rigor789** [comment]
I agree with @NathanaelA  since custom fonts are in `fonts/` folder - we should just list all fonts in the folder.

```bash
$ ns fonts

Found Custom Fonts:
-------------------
barlow-bold.ttf    ->    font-family: Barlow, barlow-bold;
fontawesome.ttf    ->    font-family: FontAwesome, fontawesome;
```

Though, I guess listing system fonts could also be useful, perhaps with a flag to reduce noise?

**Comment by NathanWalker** [comment]
I like a flag to list all if desired but default would print custom fonts in fonts folder only as long as mentions that’s the output it’s printing by default clearly and mention the flag which can be added to list all maybe / Although listing all is usually runtime related since iOS and Android only upon running can output the full list to my knowledge.

**Comment by rigor789** [comment]
Active RFC: https://github.com/NativeScript/rfcs/pull/21
